proxy/envoy/testdata/rds-httpproxy.json.golden

```diff
@@ -0,0 +1,123 @@
+{
+  "virtual_hosts": [
+   {
+    "name": "hello.default.svc.cluster.local|http",
+    "domains": [
+     "hello:80",
+     "hello",
+     "hello.default:80",
+     "hello.default",
+     "hello.default.svc:80",
+     "hello.default.svc",
+     "hello.default.svc.cluster:80",
+     "hello.default.svc.cluster",
+     "hello.default.svc.cluster.local:80",
+     "hello.default.svc.cluster.local",
+     "10.1.0.0:80",
+     "10.1.0.0"
+    ],
+    "routes": [
+     {
+      "prefix": "/",
+      "cluster": "out.e5c9564b7c4dbb0355a4f740e9d29277ccca97cd"
+     }
+    ]
+   },
+   {
+    "name": "hello.default.svc.cluster.local|http-status",
+    "domains": [
+     "hello:81",
+     "hello.default:81",
+     "hello.default.svc:81",
+     "hello.default.svc.cluster:81",
+     "hello.default.svc.cluster.local:81",
+     "10.1.0.0:81"
+    ],
+    "routes": [
+     {
+      "prefix": "/",
+      "cluster": "out.81c187d71467b1608736c57bb0734f9ef9b68f7d"
+     }
+    ]
+   },
+   {
+    "name": "httpbin.default.svc.cluster.local|http",
+    "domains": [
+     "httpbin:80",
+     "httpbin",
+     "httpbin.default:80",
+     "httpbin.default",
+     "httpbin.default.svc:80",
+     "httpbin.default.svc",
+     "httpbin.default.svc.cluster:80",
+     "httpbin.default.svc.cluster",
+     "httpbin.default.svc.cluster.local:80",
+     "httpbin.default.svc.cluster.local"
+    ],
+    "routes": [
+     {
+      "prefix": "/",
+      "host_rewrite": "httpbin.default.svc.cluster.local",
+      "cluster": "out.ae8d3361601f8293abe6ac5e4d807124612cf42e"
+     }
+    ]
+   },
+   {
+    "name": "httpsbin.default.svc.cluster.local|https",
+    "domains": [
+     "httpsbin:443",
+     "httpsbin.default:443",
+     "httpsbin.default.svc:443",
+     "httpsbin.default.svc.cluster:443",
+     "httpsbin.default.svc.cluster.local:443"
+    ],
+    "routes": [
+     {
+      "prefix": "/",
+      "host_rewrite": "httpsbin.default.svc.cluster.local",
+      "cluster": "out.242bc3028e0f3fe0682e6d972e167ab415b2321d"
+     }
+    ]
+   },
+   {
+    "name": "world.default.svc.cluster.local|http",
+    "domains": [
+     "world:80",
+     "world",
+     "world.default:80",
+     "world.default",
+     "world.default.svc:80",
+     "world.default.svc",
+     "world.default.svc.cluster:80",
+     "world.default.svc.cluster",
+     "world.default.svc.cluster.local:80",
+     "world.default.svc.cluster.local",
+     "10.2.0.0:80",
+     "10.2.0.0"
+    ],
+    "routes": [
+     {
+      "prefix": "/",
+      "cluster": "out.a2263b90cd24e500e9ed95f79ae47eabdc77dc74"
+     }
+    ]
+   },
+   {
+    "name": "world.default.svc.cluster.local|http-status",
+    "domains": [
+     "world:81",
+     "world.default:81",
+     "world.default.svc:81",
+     "world.default.svc.cluster:81",
+     "world.default.svc.cluster.local:81",
+     "10.2.0.0:81"
+    ],
+    "routes": [
+     {
+      "prefix": "/",
+      "cluster": "out.bde94496eb59ec2ed5b81392a1d32377960660b8"
+     }
```

Comment:
The route rules for HTTP are contained within the "routes", and are covered by other examples in here (see weighted RDS). We could add it here but I don't see how it can be affected by this PR.